src/app/shared/validator.service.ts

```diff
@@ -5,34 +5,35 @@ import { Observable } from 'rxjs/Observable';
 import { timer } from 'rxjs/observable/timer';
 import { fromPromise } from 'rxjs/observable/fromPromise';
 
-import { findAllDocuments } from '../shared/mangoQueries';
-import { CouchService } from '../shared/couchdb.service';
+import { MongoQueries } from './mongoQueries';
+import { CouchService } from './couchdb.service';
 
 import { switchMap, map } from 'rxjs/operators';
 
 @Injectable()
-export class NationValidatorService {
-  readonly dbName = 'nations';
+export class ValidatorService {
 
-  constructor(private couchService: CouchService) {}
+constructor(private couchService: CouchService, private mongoQueries: MongoQueries) {}
 
-  public nationCheckerService$(header: string, name: string): Observable<boolean> {
+  public CheckService$(db: string, field: string, value: string): Observable<boolean> {
     const isDuplicate = this.couchService
-      .post(`${this.dbName}/_find`, findAllDocuments(header, name))
+      .post(`${db}/_find`, this.mongoQueries.findOneDocument(field, value))
```

Comment:
Needs to be dynamic like it needs to check which function to use . It may use findOneDocument or findAllDocuments or findMultiDocuments.